### PR TITLE
Needed to get the example generator up and running

### PR DESCRIPTION
**Pull request by pdmars**
These are all things that tripped me up when I was trying to get the example generator working.

### CODE REVIEW

**Comment by TimSimpsonR** [comment]
Thanks for improving fake mode. +1

File: reddwarf/tests/fakes/nova.py [hunk 1]
```diff
@@ -290,7 +289,8 @@ def set_attachment(self, server_id):
         for attachment in self.attachments:
             if attachment['server_id'] == server_id:
                 return  # Do nothing
-        self.attachments.append({'server_id':server_id})
+        self.attachments.append({'server_id':server_id,
+                                 'device':self.mapping.device})
```
Comment (cp16net) [inline reddwarf/tests/fakes/nova.py]: i am still a little concerned about this `self.mapping.device` i dunno where mapping came from?

Comment (cp16net) [inline reddwarf/tests/fakes/nova.py]: i think this should be `self.device` instead.

**Comment by ed-** [comment]
Nice. +1

File: reddwarf/extensions/mgmt/views.py [hunk 2]
```diff
@@ -32,13 +32,16 @@ def get_ip_address(addresses):
 class InstanceView(object):
 
     def __init__(self, instance, add_addresses=False):
-        self.instance = instance
+        if isinstance(instance, list):
+            self.instance = instance[0]
+        else:
+            self.instance = instance
```
Comment (hub-cap) [inline reddwarf/extensions/mgmt/views.py]: We need to figure out why this is coming back as a list, it shouldnt be.

Comment (imsplitbit) [inline reddwarf/extensions/mgmt/views.py]: will this hold up the PR?

Comment (cp16net) [inline reddwarf/extensions/mgmt/views.py]: good catch @hub-cap

File: reddwarf/tests/fakes/nova.py (hunk 1 above)
Comment (pdmars) [inline reddwarf/tests/fakes/nova.py]: Changed.

**Comment by pdmars** [comment]
Dropped changes to the mgmt components.

**Comment by rnirmal** [comment]
+1

**Comment by cp16net** [comment]
+1

**Pull request merged**